packages/realm/src/app-services/App.ts

```diff
@@ -35,6 +35,70 @@ import {
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export type AnyApp = App<any, any>;
 
+/**
+ * This describes the persistence modes for metadata.
+ */
+export enum MetadataMode {
+  /**
+   * Persist {@link User} objects, but do not encrypt them.
+   */
+  NoEncryption = "NoEncryption",
+  /**
+   * Persist {@link User} objects in an encrypted store.
+   */
+  Encryption = "Encryption",
+  /**
+   * Do not persist {@link User} objects.
+   */
+  NoMetadata = "NoMetadata",
+}
```

Comment:
I think we at some point tried to decide on a convention for all new public-facing enum constants. I think the most used in our code base is `camelCase` and `kebab-case`. I believe we opted for `kebab-case`. If we were to modify the ones here for `MetadataMode`, it wouldn't require any refactors elsewhere in the code right? (Doesn't look like that at first glance anyway.)

Unsure of what the others think here, but I'd say that it would probably be good to try to use one of the existing casing styles for the enum constants.